recipe/utf_8_mac.patch

```diff
@@ -0,0 +1,1737 @@
++ * formats and network protocols.  unicode_combine will combine
++ * a decomposed character sequence into a single precomposed
++ * (composite) character.
++ *
++ * Similar toCFUniCharPrecomposeCharacter but unicode_combine
++ * also handles Hangul Jamo characters.
++ */
++static u_int16_t
++unicode_combine(u_int16_t base, u_int16_t combining)
++{
++	u_int32_t value;
++
++	/* Check HANGUL */
++	if ((combining >= HANGUL_VBASE) && (combining < (HANGUL_TBASE + HANGUL_TCOUNT))) {
++		/* 2 char Hangul sequences */
++		if ((combining < (HANGUL_VBASE + HANGUL_VCOUNT)) &&
++		    (base >= HANGUL_LBASE && base < (HANGUL_LBASE + HANGUL_LCOUNT))) {
++		    return (HANGUL_SBASE +
++		            ((base - HANGUL_LBASE)*(HANGUL_VCOUNT*HANGUL_TCOUNT)) +
++		            ((combining  - HANGUL_VBASE)*HANGUL_TCOUNT));
++		}
++	
++		/* 3 char Hangul sequences */
++		if ((combining > HANGUL_TBASE) &&
++		    (base >= HANGUL_SBASE && base < (HANGUL_SBASE + HANGUL_SCOUNT))) {
++			if ((base - HANGUL_SBASE) % HANGUL_TCOUNT)
++				return (0);
++			else
++				return (base + (combining - HANGUL_TBASE));
++		}
++	}
++
++	value = getmappedvalue32(
++		(const unicode_mappings32 *)__CFUniCharPrecompSourceTable,
++		__CFUniCharPrecompositionTableLength, combining);
++
++	if (value) {
++		value = getmappedvalue16(
++			(const unicode_mappings16 *)
++			((u_int32_t *)__CFUniCharBMPPrecompDestinationTable + (value & 0xFFFF)),
++			(value >> 16), base);
++	}
++	return (value);
++}
++
++static int
++utf8mac_mbtowc (conv_t conv, ucs4_t *pwc, const unsigned char *s, size_t n)
++{
++    u_int16_t ucsp[13];
++    size_t ucslen = 0, consumed = 0;
++    int ret;
++    int flags;
++
++    bzero(&ucsp, sizeof(ucsp));
++    *pwc = 0;
++    flags = UTF_PRECOMPOSED;
++
++#ifdef __LITTLE_ENDIAN__
++    flags |= UTF_REVERSE_ENDIAN;
++#endif
++
++    ret = utf8_decodestr(s, n, ucsp, &ucslen, sizeof(ucsp), 0, flags, &consumed);
++
++    if (ret == ENAMETOOLONG)	/* Name didn't fit; only ucslen chars were decoded */
++	return RET_TOOFEW(0);
++
++    if ( ret == EINVAL)		/* Illegal UTF-8 sequence found */
++	return RET_ILSEQ;
++
++    if((ret = _utf16be_mbtowc(pwc, (const unsigned char *) ucsp, ucslen)) < 0)
++        return ret;
++
++    return (int)consumed;
++}
++
++static int
++utf8mac_wctomb (conv_t conv, unsigned char *r, ucs4_t wc, size_t n) /* n == 0 is acceptable */
++{
++    int ret;
++    size_t len;
++    u_int16_t ucs_string[13];
++    int flags;
++
++    if((ret = _utf16be_wctomb((unsigned char *) ucs_string, wc, sizeof(ucs_string))) < 0)
++        return ret;
++
++    flags = UTF_NO_NULL_TERM | UTF_DECOMPOSED;
++#ifdef __LITTLE_ENDIAN__
++    flags |= UTF_REVERSE_ENDIAN;
++#endif
++    ret = utf8_encodestr(ucs_string, ret, r, &len, n, 0, flags);
++    if (ret == EINVAL)
++        return RET_ILUNI;
++    if (ret == ENAMETOOLONG)
++        return RET_TOOSMALL;
++
++    return (int)len;
++}
++
++/* vim:set tabstop=8 shiftwidth=4: */
```

Comment:
We could use the latest source from https://github.com/apple-oss-distributions/libiconv/blob/libiconv-64/libiconv/lib/utf8mac.h
which is from https://github.com/apple-oss-distributions/libiconv/commit/d5939135328109e5f591a4fa71faf71a24aa3917.patch
(alternatively we could also add this as an additional source with `url: https://raw.githubusercontent.com/apple-oss-distributions/libiconv/libiconv-64/libiconv/lib/utf8mac.h, folder: lib}`)

Reply:
I'd prefer not to pull the patch dynamically just yet.

Reply:
Oh, my main suggestion was just to update `utf_8_mac.patch` with the updated source.
Totally fine by me to have the source hard-coded in the patch file.